src/main/java/seedu/address/model/assignment/Assignment.java

```diff
@@ -1,14 +1,55 @@
 package seedu.address.model.assignment;
 
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import seedu.address.commons.util.ToStringBuilder;
+
 /**
- * TODO implement this class
+ * Represents a person's assignment
+ * Guarantees: immutable; fields are non-null except for grade
  */
 public class Assignment {
 
+    private AssignmentName assignmentName;
+    private Deadline deadline;
+    private Status submissionStatus;
+    private Status gradingStatus;
+    private Grade grade;
+
+    /**
+     * Constructs a {@code Assignment}
+     *
+     * @param assignmentName A valid name
+     * @param deadline A valid date
+     * @param submissionStatus A valid subsmission status
+     * @param gradingStatus A valid grading status
+     * @param grade A valid grade
+     */
+    public Assignment(AssignmentName assignmentName, Deadline deadline, Status submissionStatus,
+                      Status gradingStatus, Grade grade) {
+        requireAllNonNull(assignmentName, deadline, submissionStatus, gradingStatus);
+        this.assignmentName = assignmentName;
+        this.deadline = deadline;
+        this.submissionStatus = submissionStatus;
+        this.gradingStatus = gradingStatus;
+        this.grade = grade;
```

Comment:
I was thinking should we make filling up the grade a non-compulsory field initially when you are creating assignments for students. 🤔

Reply:
Nvm, I noticed that you can pass null as a value, but this could just be a possible consideration.

Reply:
Edited so that if left empty, the fields default to unsubmitted, ungraded and no grade respectively